pyteal/ast/abi/int.py

```diff
@@ -0,0 +1,185 @@
+from typing import (
+    TypeVar,
+    Union, cast
+)
+from abc import abstractmethod
+
+from pyteal.ast.assert_ import Assert
+from pyteal.ast.naryexpr import And
+from pyteal.ast.seq import TealInputError, Seq
+from pyteal.ast.int import Int
+from pyteal.ast.expr import Expr
+from pyteal.ast.abi.type import ComputedValue
+from pyteal.ast.abi.uint import Uint, UintTypeSpec
+
+
+class SIntTypeSpec(UintTypeSpec):
+    @abstractmethod
+    def new_instance(self) -> "SInt":
+        pass
+
+    @abstractmethod
+    def annotation_type(self) -> "type[SInt]":
+        pass
+
+    def __str__(self) -> str:
+        return "sint{}".format(self.bit_size())
+
+
+SIntTypeSpec.__module__ = "pyteal"
+
+
+class SInt8TypeSpec(SIntTypeSpec):
+    def __init__(self):
+        super().__init__(8)
+
+    def new_instance(self) -> "SInt8":
+        return SInt8()
+
+    def annotation_type(self) -> "type[SInt8]":
+        return SInt8
+
+
+SInt8TypeSpec.__module__ = "pyteal"
+
+
+class SInt16TypeSpec(SIntTypeSpec):
+    def __init__(self):
+        super().__init__(16)
+
+    def new_instance(self) -> "SInt16":
+        return SInt16()
+
+    def annotation_type(self) -> "type[SInt16]":
+        return SInt16
+
+
+SInt16TypeSpec.__module__ = "pyteal"
+
+
+class SInt32TypeSpec(SIntTypeSpec):
+    def __init__(self):
+        super().__init__(32)
+
+    def new_instance(self) -> "SInt32":
+        return SInt32()
+
+    def annotation_type(self) -> "type[SInt32]":
+        return SInt32
+
+
+SInt32TypeSpec.__module__ = "pyteal"
+
+
+class SInt64TypeSpec(SIntTypeSpec):
+    def __init__(self):
+        super().__init__(64)
+
+    def new_instance(self) -> "SInt64":
+        return SInt64()
+
+    def annotation_type(self) -> "type[SInt64]":
+        return SInt64
+
+
+SInt64TypeSpec.__module__ = "pyteal"
+
+
+T = TypeVar("T", bound="Int")
```

Comment:
I think this should be `SInt`
```suggestion
T = TypeVar("T", bound="SInt")
```